build.xml

```diff
@@ -22,18 +23,339 @@
+	<target name="init-core" description="Prepare project before build">
+		<delete dir="${build-project.dir}"/>
+		<mkdir dir="${build-project.dir}"/>
+		<mkdir dir="${build-project.src.dir}"/>
+		<mkdir dir="${build-project.test.dir}"/>
+		<mkdir dir="${build-project.log.dir}"/>
+		<copy includeemptydirs="false" todir="${build-project.src.dir}">
+			<fileset dir="src">
+				<exclude name="**/*.launch"/>
+				<exclude name="**/*.xtend"/>
+				<exclude name="**/*.java"/>
+			</fileset>
+		</copy>
+		<copy includeemptydirs="false" todir="${build-project.test.dir}">
+			<fileset dir="test">
+				<exclude name="**/*.launch"/>
+				<exclude name="**/*.xtend"/>
+				<exclude name="**/*.java"/>
+			</fileset>
 		</copy>
 	</target>
 
-	<target name="tomcat_clean_static_content" description="Delete static content">
-		<delete dir="${env.CATALINA_HOME}/work/Catalina/localhost"></delete>
+	<target name="build-core" description="Build Scada-LTS core application">
+		<record action="start" name="${build-project.log.src.name}"/>
+		<echo message="${ant.project.name}: ${ant.file}"/>
+		<javac debug="true" debuglevel="${debuglevel}" destdir="${build-project.src.dir}" includeantruntime="false" source="${source}" target="${target}">
+			<src path="src"/>
+			<classpath refid="scada-lts.classpath"/>
+		</javac>
+		<record action="stop" name="${build-project.log.src.name}"/>
+		<record action="start" name="${build-project.log.test.name}"/>
+		<javac debug="true" debuglevel="${debuglevel}" destdir="${build-project.test.dir}" includeantruntime="false" source="${source}" target="${target}">
+			<src path="test"/>
+			<classpath refid="scada-lts.classpath"/>
+			<classpath refid="test-unit.classpath"/>
+			<classpath refid="benchmark.classpath"/>
+		</javac>
+		<record action="stop" name="${build-project.log.test.name}"/>
 	</target>
 
-	<target name="run" depends="tomcat_clean_static_content">
+	<target name="init-static-dependencies" description="Install JavaScript dependencies for Scada-LTS using nodejs.">
+		<exec dir="./WebContent/resources" executable="npm.cmd" osfamily="windows">
+			<arg value="install"/>
+		</exec>
+		<exec dir="./WebContent/resources" executable="npm" osfamily="unix">
+			<arg value="install"/>
+		</exec>
+	</target>
+
+	<target name="init-ui-dependencies" description="Install frontend application dependencies">
+		<exec dir="./scadalts-ui" executable="npm.cmd" osfamily="windows">
+			<arg value="install"/>
+		</exec>
+		<exec dir="./scadalts-ui" executable="npm" osfamily="unix">
+			<arg value="install"/>
+		</exec>
+	</target>
+
+	<target name="build-ui" description="Build frontend application">
+		<exec dir="./scadalts-ui" executable="npm.cmd" osfamily="windows">
+			<arg value="run"/>
+			<arg value="build"/>
+		</exec>
+		<exec dir="./scadalts-ui" executable="npm" osfamily="unix">
+			<arg value="run"/>
+			<arg value="build"/>
+		</exec>
+	</target>
+
+	<target name="test-junit" description="Test Scada-LTS core application">
```

Comment:
depends="build-core"

Reply:
`test-junit` is a basic task that is a part of the `build-no-ui` task group.  
Basic task should not depends on any task. To perform 'test-junit' that depends on `"build-core"` use `"build-no-ui"`.
See the build.xml file because it is already done in configuration. Description is provied in "readme.md" file and in the PullRequest description.

Reply:
When you run tests, you usually want to check the current state of the code. Similarly, here I propose a small compromise between the atomicity of these targets, and that the essential ones should be autonomous.